OMEdit/OMEditLIB/OMS/ElementPropertiesDialog.cpp

```diff
@@ -403,3 +405,89 @@ void ElementPropertiesDialog::updateProperties()
+        updateStartValues(pParameterLabelText, pLineEdit);
+      }
+
+      if (inputIndex != -1) {
+        QString pInputLabelText = mInputLabels.at(inputIndex)->text();
+        updateStartValues(pInputLabelText, pLineEdit);
+      }
+    }
+  }
+  return QWidget::eventFilter(pObject, pEvent);
+}
+
+/*
+ * helper function to updateStartValues read from modeldescription.xml
+ */
+void ElementPropertiesDialog::updateStartValues(QString pName, QLineEdit * pLineEdit)
+{
+  if (mpComponent->getLibraryTreeItem()->getOMSElement() && mpComponent->getLibraryTreeItem()->getOMSElement()->connectors) {
+    oms_connector_t** pInterfaces = mpComponent->getLibraryTreeItem()->getOMSElement()->connectors;
+    for (int i = 0 ; pInterfaces[i] ; i++) {
+      QString nameStructure = QString("%1.%2").arg(mpComponent->getLibraryTreeItem()->getNameStructure(), QString(pInterfaces[i]->name));
+      if (pInterfaces[i]->causality == oms_causality_parameter && QString(pInterfaces[i]->name)== pName) {
+        OMSProxy::instance()->omsDelete(nameStructure + ":start");
+        updateStartValuesHelper(pInterfaces[i], nameStructure, pLineEdit);
+        break;
+      } else if (pInterfaces[i]->causality == oms_causality_input && QString(pInterfaces[i]->name)== pName) {
+        OMSProxy::instance()->omsDelete(nameStructure + ":start");
+        updateStartValuesHelper(pInterfaces[i], nameStructure, pLineEdit);
+        break;
+      }
+    }
+  }
+}
+
+/*
+ * helper function to updateStartValues read from modeldescription.xml
+ */
+void ElementPropertiesDialog::updateStartValuesHelper(oms_connector_t* pConnector, QString pName, QLineEdit *pLineEdit)
+{
+  bool status = false;
+  if (pConnector->type == oms_signal_type_real) {
+    double value;
+    if ((status = OMSProxy::instance()->getReal(pName, &value))) {
+      pLineEdit->setText(QString::number(value));
+    }
+  } else if (pConnector->type == oms_signal_type_integer) {
+    int value;
+    if ((status = OMSProxy::instance()->getInteger(pName, &value))) {
+      pLineEdit->setText(QString::number(value));
+    }
+  } else if (pConnector->type == oms_signal_type_boolean) {
+    bool value;
+    if ((status = OMSProxy::instance()->getBoolean(pName, &value))) {
+      pLineEdit->setText(QString::number(value));
+    }
+  } else if (pConnector->type == oms_signal_type_string) {
+    qDebug() << "ElementPropertiesDialog::ElementPropertiesDialog() oms_signal_type_string not implemented yet.";
+  } else if (pConnector->type == oms_signal_type_enum) {
+    qDebug() << "ElementPropertiesDialog::ElementPropertiesDialog() oms_signal_type_enum not implemented yet.";
+  } else if (pConnector->type == oms_signal_type_bus) {
+    qDebug() << "ElementPropertiesDialog::ElementPropertiesDialog() oms_signal_type_bus not implemented yet.";
+  } else {
+    qDebug() << "ElementPropertiesDialog::ElementPropertiesDialog() unknown oms_signal_type_enu_t.";
+  }
+  if (!status) {
+    pLineEdit->setPlaceholderText("unknown");
```

Comment:
When and where does this pop up?

Reply:
this will pop when the signal type is any one of the following 

1. oms_signal_type_string
2. oms_signal_type_enum
3. oms_signal_type_bus